src/Samples/SubModelSeq/Program.fs

```diff
@@ -4,6 +4,24 @@ open System
 open Elmish
 open Elmish.WPF
 
+
+let mapIf p f a =
+  if p a then f a else a
+
+
+module Func =
+
+  let flip<'a, 'b, 'c> (f: 'a -> 'b -> 'c) =
+    fun b a -> f a b
```

Comment:
This can be simplified to

```suggestion
  let flip f a b = f b a
```

but given that it's only used a single place, I suggest just removing it and rewriting the relevant code.

Reply:
I prefer to keep it as a separate function.  Haskell considers it so important that [it is include in their prelude](http://zvon.org/other/haskell/Outputprelude/flip_f.html).  I wish F# thought as highly about this function.

If the arguments appear in this order `let flip f b a = f a b`, then the inferred types match the argument names.  I have changed to that syntax and will leave this conversation for you to resolve or continue.

Reply:
> Haskell considers it so important that [it is include in their prelude](http://zvon.org/other/haskell/Outputprelude/flip_f.html). I wish F# thought as highly about this function.

I agree, though I always use FSharpPlus which adds it, so it's a moot point for me.

In any case the reason I suggested removing it here was to simplify the sample (less to think about), but I'm fine with keeping it, especially with the rewritten syntax.